$MirOS: ports/converters/rpm2cpio/patches/patch-rpm2cpio,v 1.1.7.1 2005/03/18 15:44:05 tg Exp $
--- rpm2cpio.orig	Sun Sep 12 14:39:53 2004
+++ rpm2cpio	Sat Aug 28 15:43:11 2004
@@ -1,47 +1,37 @@
 #!/usr/bin/perl
-
+# $MirOS: ports/converters/rpm2cpio/patches/patch-rpm2cpio,v 1.1.7.1 2005/03/18 15:44:05 tg Exp $
+#
 # Copyright (C) 1997,1998,1999, Roger Espel Llima
-# 
+#
 # Permission is hereby granted, free of charge, to any person obtaining a copy
-# of this software and any associated documentation files (the "Software"), to 
-# deal in the Software without restriction, including without limitation the 
-# rights to use, copy, modify, merge, publish, distribute, sublicense, 
-# and/or sell copies of the Software, and to permit persons to whom the 
+# of this software and any associated documentation files (the "Software"), to
+# deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
 # Software is furnished to do so, subject to the following conditions:
-# 
+#
 # The above copyright notice and this permission notice shall be included in
 # all copies or substantial portions of the Software.
-# 
+#
 # THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 # IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 # FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
-# SOFTWARE'S COPYRIGHT HOLDER(S) BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER 
+# SOFTWARE'S COPYRIGHT HOLDER(S) BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 # LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 # THE SOFTWARE
 
-# (whew, that's done!)
-
-# why does the world need another rpm2cpio?  because the existing one
-# won't build unless you have half a ton of things that aren't really
-# required for it, since it uses the same library used to extract RPM's.
-# in particular, it won't build on the HPsUX box i'm on.
-
-# sw 2002-Mar-6 Don't slurp the whole file
-
-# add a path if desired
-$gzip = "gzip";
-
 sub printhelp {
-  print <<HERE;
-rpm2cpio, perl version by orabidoo <odar\@pobox.com> +sw
+  print STDERR <<HERE;
+rpm2cpio, perl version by orabidoo <odar\@pobox.com> +sw for MirOS
 dumps the contents to stdout as a cpio archive
 
-use: rpm2cpio [file.rpm] > file.cpio
+use: rpm2cpio [file.rpm] >file.cpio
 
 Here's how to use cpio:
-     list of contents:   cpio -t -i < /file/name
-        extract files:   cpio -d -i < /file/name
+     list of contents:   tar tf file.cpio
+        extract files:   tar xp[hv]f file.cpio
+You could use "cpio -idm[v] <file.cpio" as well...
 HERE
 
   exit 0;
@@ -59,12 +49,6 @@ if ($#ARGV == -1) {
 
 printhelp if -t STDOUT;
 
-# gobble the file up
-##undef $/;
-##$|=1;
-##$rpm = <$f>;
-##close ($f);
-
 read $f,$rpm,96;
 
 ($magic, $major, $minor, $crap) = unpack("NCC C90", $rpm);
@@ -72,8 +56,6 @@ read $f,$rpm,96;
 die "Not an RPM\n" if $magic != 0xedabeedb;
 die "Not a version 3 or 4 RPM\n" if $major != 3 && $major != 4;
 
-##$rpm = substr($rpm, 96);
-
 while (!eof($f)) {
   $pos = tell($f);
   read $f,$rpm,16;
@@ -99,7 +81,7 @@ if (eof($f)) {
   die "bogus RPM\n";
 }
 
-open(ZCAT, "|gzip -cd") || die "can't pipe to gzip\n";
+open(ZCAT, "|/usr/bin/gzip -cd") || die "can't pipe to gzip\n";
 print STDERR "CPIO archive found!\n";
 
 print ZCAT $rpm;
@@ -109,4 +91,3 @@ while (read($f, ($_=''), 16384) > 0) {
 }
 
 close ZCAT;
-
